cpp/euler/primes.h

```diff
@@ -0,0 +1,83 @@
+#include <vector>
+#include <sstream>
+#include <iostream>
+
+#define ll long long
+#define ull unsigned long long
+
+/**
+ * Helpful Links :
+ * 1. https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes
+ * 2. https://www.cplusplus.com/reference/sstream/ostringstream/
+ */
+class SieveOfEratosthenes {
+private:
+  std::vector<bool> sieve;
+  std::vector<int> primes;
+
+  void initializeSieve() {
+    sieve[0] = sieve[1] = false;
+    for (unsigned candidate = 2; candidate < sieve.size(); candidate++) {
+      if (sieve[candidate]) {
+        primes.push_back(candidate);
+        unsigned consonant = candidate + candidate;
+        while(consonant < sieve.size()) {
+          sieve[consonant] = false;
+          consonant += candidate;
+        }
+      }
+    }
+  }
+public:
+  SieveOfEratosthenes(size_t max) {
+    this->sieve = *(new std::vector<bool>(max + 1, true));
```

Comment:
Why not use the constructor initializer with a colon after the constructor?

Something like: 

SieveOfE(int max) : sieve(max+1, true), primes(std::max(max/log(max),1024)) {...}